src/Source/CopcSource.js

```diff
@@ -102,8 +103,25 @@ class CopcSource extends Source {
             this.header = metadata.header;
             this.info = metadata.info;
             this.eb = metadata.eb;
-            // TODO: use wkt definition in `metadata.wkt` to infer/define crs
-            this.crs = config.crs || 'EPSG:4326';
+
+            proj4.defs('unknown', metadata.wkt);
+
+            if (proj4.defs('unknown').type === 'COMPD_CS') {
+                console.warn('CopcSource: compound coordinate system is not yet supported.');
+                metadata.wkt = metadata.wkt.slice(metadata.wkt.search('PROJCS'), metadata.wkt.search(',VERT_CS'));
+                proj4.defs('unknown', metadata.wkt);
+            }
+
+            const projCS = proj4.defs('unknown');
+            if (projCS.AUTHORITY) {
+                const authority = Object.keys(projCS.AUTHORITY)[0];
+                this.crs = `${authority}:${projCS.AUTHORITY[authority]}`;
+                if (!proj4.defs(this.crs)) {
+                    proj4.defs(this.crs, proj4.defs('unknown'));
+                }
+            } else {
+                this.crs = projCS.name || 'EPSG:4326';
+            }
```

Comment:
Seems we are patching a missing feature of `proj4.js`, I'm for proposing a PR directly upstream instead of patching it in `iTowns`...

Reply:
I did open a PR on proj4 github:
https://github.com/proj4js/wkt-parser/pulls

Reply:
I simplified it and if ever proj4 accept my PR or a version of it, we might have to change one line in the test copc.js

Reply:
PR in proj4 merged. (I don't know about the deployment of it)